lua/telescope/make_entry.lua

```diff
@@ -283,6 +283,77 @@ function make_entry.gen_from_quickfix(opts)
   end
 end
 
+function make_entry.gen_from_symbols(opts)
+  opts = opts or {}
+  opts.tail_path = get_default(opts.tail_path, true)
+
+    local displayer = entry_display.create {
+    -- separator = "▏",
+    separator = " ",
+    items = {
+      { width = 6 },
+      { width = 40 },
+      { width = 1 },
+      { remaining = true },
+      { width = 1 },
+    },
+  }
+
+  local make_display = function(entry)
+    local filename
+    if not opts.hide_filename then
+      filename = entry.filename
+      if opts.tail_path then
+        filename = utils.path_tail(filename)
+      elseif opts.shorten_path then
+        filename = utils.path_shorten(filename)
+      end
+    end
+
+    local type_highlight = {
+      ["Constant"] = "Constant",
+      ["Field"]    = "Function",
+      ["Function"] = "Function",
+      ["Property"] = "Operator",
+      ["Variable"] = "SpecialChar",
+    }
+
+    return displayer {
+      {entry.lnum .. ":" .. entry.col, "LineNr"},
+      entry.symbol_name,
+      {"[", "TelescopeBorder"},
+      {entry.symbol_type, type_highlight[entry.symbol_type],type_highlight[entry.symbol_type]},
```

Comment:
`type_highlight[entry.symbol_type]` is here twice

Reply:
that was for the "override match colors" . I'm not sure that it's needed here, but that's what was the intention.

Reply:
Oh i have not jet implemented that. So right now i only process the first element(string) and the second element hlgroup. But if highlighting fix is about to get merged we can think about this.